c_plus_plus/src/model/audio_suppresser/audio_suppresser.cpp

```diff
@@ -0,0 +1,52 @@
+#include "audio_suppresser.h"
+
+namespace Model
+{
+
+void AudioSuppresser::suppressSources(std::vector<int> sourcesToSuppress, uint8_t* audioBuf, const int bufferLength)
+{
+    // Initialize mask
+    uint8_t mask[bufferLength];
+
+    for(int i = 0; i < bufferLength; i++)
+    {
+        mask[i] = 255;
+    }
+
+    // Build mask
+    int index;
+    for(std::size_t i = 0; i < sourcesToSuppress.size(); i++)
+    {
+        index = sourcesToSuppress[i];
+        switch (index)
+        {
+        case 0: createMaskFromIndex(static_cast<int>(index), mask, bufferLength);
+                break;
+        case 1: createMaskFromIndex(static_cast<int>(index), mask, bufferLength);
+                break;
+        case 2: createMaskFromIndex(static_cast<int>(index), mask, bufferLength);
+                break;
+        case 3: createMaskFromIndex(static_cast<int>(index), mask, bufferLength);
+                break;
+        }
```

Comment:
why there is a switch case if each case does the same thing?

Reply:
And index is already an integer, why we added a static_cast < int > ?

Reply:
Oops, I forgot to remove it. Good catch.